### DISCUSSION TITLE
Using UUIDv5 as primary key for Bookmark records

### DISCUSSION THREAD

**Discussion by le-krogoth** [go-shiori/shiori#701] [Feature requests]
I have a use case where I have two and possible more Shiori databases on different hosts and want to regularly merge those to get one single archive. Merging databases with sequences as primary keys is difficult if not impossible. At least in an automated fashion.

I looked at the database schema of Shiori and wondered if you would be interested to change the primary key of the bookmark table from int to UUID v5? The UUID could be generated from the URL, so would always be the same for the same URL (as long as one uses the same namespace). And having a UUID would definitely help when merging several Shiori databases.

Whats your take on that?

**Comment by fmartingr** [comment]
It's something that I have on radar but I can't invest my time on at the moment, there would be work to be done in order to make it properly backwards compatible (since there are public archive links involved) and we currently can't run arbitrary code on migrations.

For your use case, if you don't mind ID changes, you can merge your databases removing the ID from the `INSERT` statements, that would just keep using the next one on the database you merge records to.
